sharding/collation.go

```diff
@@ -122,9 +122,10 @@ func (c *Collation) CalculateChunkRoot() {
 // some salt, which is appended to each chunk in the collation body before it
 // is hashed.
 func (c *Collation) CalculatePOC(salt []byte) common.Hash {
-	body := make([]byte, len(c.body))
+	body := make([]byte, 0, len(c.body)*(1+len(salt)))
+
 	for _, chunk := range c.body {
-		body = append(body, append(salt, chunk)...) // add salt to each chunk.
+		body = append(append(body, salt...), chunk)
```

Comment:
Can you explain the logic behind `append(append(body, salt...), chunk)`? Is each chunk 32 byte?

Reply:
Yeah, good catch. A chunk should be a 32 byte array but here it's obviously just a byte. A more correct version of this would iterate over every 32 bytes of `c.body` and use that as `chunk` instead. 

Here I just prepended `chunk` with `salt` because the scope of this PR was to make the existing method run faster, not to change the behavior.

Reply:
That's fair. Can you add a TODO comment on top of append so we don't miss it later, I will open an issue as well
Something like `TODO: use every 32bytes of c.body as chunk`

Reply:
got it!